src/vivarium/interface/interactive.py

```diff
@@ -118,39 +131,51 @@ def run_until(self, end_time: Time, with_logging: bool = True) -> int:
 
         """
         if not isinstance(end_time, type(self._clock.time)):
-            raise ValueError(f"Provided time must be an instance of {type(self._clock.time)}")
-
-        iterations = int(ceil((end_time - self._clock.time) / self._clock.step_size))
-        self.take_steps(number_of_steps=iterations, with_logging=with_logging)
-        assert self._clock.time - self._clock.step_size < end_time <= self._clock.time
-        return iterations
-
-    def take_steps(
-        self, number_of_steps: int = 1, step_size: Timedelta = None, with_logging: bool = True
-    ):
-        """Run the simulation for the given number of steps.
+            msg = f"Provided time must be an instance of {type(self._clock.time)}"
+            raise TypeError(msg)
+
+        step_size = step_size if step_size is not None else self._clock.step_size
+        number_of_steps = int(ceil((end_time - self._clock.time) / self._clock.step_size))
```

Comment:
Would it make sense to make `clock.steps_remaining` a function that takes an optional parameter of end time? This is the same logic that you are using there.

Reply:
I don't think so.  I think of the clock as a closed system with a current time and step size and some functions for advancing time, and not really a calculator for time-based inquiries.  But I'm not adamant about it.

Reply:
As a compromise, you could have a utility function in the time module that is called in both locations?